code/datums/traits/good.dm

```diff
@@ -41,6 +41,22 @@
 	mob_trait = TRAIT_EMPATH
 	gain_text = "<span class='notice'>You feel in tune with those around you.</span>"
 	lose_text = "<span class='danger'>You feel isolated from others.</span>"
+	
+/datum/quirk/Supportanimal
+	name = "Emotional Support Animal"
+	desc = "You gain a random pet delivery beacon give it lots of love and care and it will take care of your mood"
+	value = 1
+	gain_text = "<span class='notice'>You feel happy with your pet.</span>"
+	lose_text = "<span class='danger'>You lament the loss of your pet.</span>"
+
+/datum/quirk/Supportanimal/on_spawn()
+	var/mob/living/carbon/human/H = quirk_holder
```

Comment:
What type is quirk holder?

Reply:
stare at this /code/datums/traits/_quirk.dm